Description: Build and configuration changes for Debian platforms
 * Define MAXPATHLEN, PATH_MAX, MAXHOSTNAMELEN, MAX_IOV for Hurd.
 * Disable libevent's epoll backend on Linux as calling epoll() on
   /dev/null hangs.
 * Disable cwd detection on FreeBSD since it requires the FreeBSD
   version of libutil.
Author: Karl Ferdinand Ebert <kfebert@gmail.com>
Author: Romain Francoise <rfrancoise@debian.org>
Bug-Debian: http://bugs.debian.org/609333
Bug-Debian: http://bugs.debian.org/631984
Forwarded: not-needed

--- a/tmux.h
+++ b/tmux.h
@@ -397,6 +397,22 @@
 	char		argv[COMMAND_LENGTH];
 };
 
+#ifndef MAXPATHLEN
+#define MAXPATHLEN 4096
+#endif
+
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
+
+#ifndef MAXHOSTNAMELEN
+#define MAXHOSTNAMELEN 64
+#endif
+
+#ifndef IOV_MAX
+#define IOV_MAX	1024
+#endif
+
 struct msg_identify_data {
 	char		cwd[MAXPATHLEN];
 
--- a/osdep-freebsd.c
+++ b/osdep-freebsd.c
@@ -29,7 +29,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#if 0
 #include <libutil.h>
+#endif
 
 struct kinfo_proc	*cmp_procs(struct kinfo_proc *, struct kinfo_proc *);
 char			*osdep_get_name(int, char *);
@@ -135,6 +137,7 @@
 char *
 osdep_get_cwd(pid_t pid)
 {
+#if 0
 	static char		 wd[PATH_MAX];
 	struct kinfo_file	*info = NULL;
 	int			 nrecords, i;
@@ -151,6 +154,7 @@
 	}
 
 	free(info);
+#endif
 	return (NULL);
 }
 
--- a/osdep-linux.c
+++ b/osdep-linux.c
@@ -88,6 +88,6 @@
 	 * poll or select, causing hangs.
 	 * 
 	 */
-	/* setenv("EVENT_NOEPOLL", "1", 1); */
+	setenv("EVENT_NOEPOLL", "1", 1);
 	return (event_init());
 }
